gcsfs/core.py

```diff
@@ -501,7 +504,16 @@ def exists(self, path):
             if key:
                 return bool(self.info(path))
             else:
-                return bucket in self.ls('')
+                if bucket in self.ls(''):
+                    return True
+                else:
+                    try:
+                        self._list_bucket(bucket)
+                        return True
+                    except:
```

Comment:
We probably want to avoid bare `except` blocks.  They also catch things like `KeyboardInterrupts`.  At the very least we should do `except Exception:`